### DISCUSSION TITLE
WebviewView support in theia

### DISCUSSION THREAD

**Discussion by chakraborty-slx** [eclipse-theia/theia#9794] [Q&A]
Hi all,

I am planning to migrate to Theia IDE, but i have questions regarding the compatibility, I have already checked this page 
https://che-incubator.github.io/vscode-theia-comparator/status.html

Q1 - Can anyone point me out how can I come to know which apis are planned for the next release for example "WebviewView".
I believe there must be a roadmap for adoptions of vscode apis?

If I want to adapt that in THEIA , where should I start?

Q2 - Can I use my vscode extension as browser / electron frontend extension and write a theia backend extension, communicate between them using JSON-RPC over sockets. As if today my vscode extension uses JSON-RPC over IO to connect to a python-based server which is based on LSP protocol. I want to use my theia backend extension to connect to the python-based server . Any example similar will be helpful.

Regards,
Abhishek Chakraborty

**Comment by msujew** [comment]
Hi @chakraborty-slx, thank you for your question.

Q1: Theia does not really have a roadmap for adopting vscode APIs, as most things are done on demand. Usually the `vscode` [issue label](https://github.com/eclipse-theia/theia/issues?q=is%3Aissue+is%3Aopen+label%3Avscode) is used to mark any issues related to that.

Adding new vscode APIs is in my experience best done bottom-up. Add the necessary APIs to the [`typescript definition`](https://github.com/eclipse-theia/theia/blob/master/packages/plugin/src/theia.d.ts) and go up from there, passing it through to any service as necessary until you are able to display it in the frontend. I'm not really familiar enough with the `WebviewView` API to say anything more about that.

Q2: You're probably overthinking this: vscode extensions are running always in Theia's backend, only surfacing some information to the frontend, like commands, problem markers, webviews, and so on. Therefore, JSON-RPC over IO connections are still working as usual, and there's no need to rewrite them for Theia. Do you still see a special reason to write a dedicated Theia backend extensions for that?

**Reply by chakraborty-slx** [reply]
@msujew thanks for your reply, I will look more into those vscode issue label to see the steps for adopting weviewview api from vscode.

A Dedicated Theia backend extension might not be required if the VSCode extension is treated as a backend, this is really helpful .

**Reply by azatsarynnyy** [reply]
> I believe there must be a roadmap for adoptions of vscode apis?

@chakraborty-slx we have no roadmap specific to VS Code APIs, but general Theia roadmap you can find here https://github.com/eclipse-theia/theia/wiki/Eclipse-Theia-Roadmap
